base/gcutils.jl

```diff
@@ -49,15 +49,29 @@ Module with garbage collection utilities.
 """
 module GC
 
+# @enum-like structure
+struct CollectionType
```

Comment:
Out of curiosity, what is the reason for not using an actual `@enum`?

Reply:
`gcutils.jl` is included very early in the system image, before `Enums.jl` and having `@enum`.